internal/agent/hooks/gruboptions.go

```diff
@@ -17,6 +18,19 @@ func (b GrubOptions) Run(c config.Config) error {
 	return nil
 }
 
+// KRun is a temporary function that does the same as Run. It will be removed as soon as the transition from config.Config to schema.KConfig is finished.
+func (b GrubOptions) KRun(kc schema.KConfig) error {
```

Comment:
I'm trying to understand the context - it is hard to follow up with dups methods. Why the duplication? can't we try this directly modifying the configurations here?

Reply:
yeah sorry, wasn't sure what the best approach for the refactoring was. I wanted to switch directly but there's one dependency in [kcrypt](https://github.com/kairos-io/kcrypt/blob/fa9eb79998b3642576ac26d49c133bbb9d372f1f/pkg/config/config.go#L50) so in the end I decided to go with the duplicated methods. But I don't mind approaching it differently if you think is best it just felt easier to have both for a while, switch everything including kcrypt and then end up with just one version of them again